### ISSUE TITLE
Enhanced Handling of Gift Code Status and Proxy Support

### ISSUE THREAD

**Issue by ed6dev** [logicguy1/Discord-Nitro-Generator-and-Checker#177]
**Description:**

I've reviewed the code and noticed that the tool currently relies solely on the 200 HTTP status code when checking gift codes. This approach can be improved for a more accurate assessment of gift codes' validity and status.

Here are the proposed enhancements:

1. **Gift Code Status:** The tool should consider checking the JSON response from the API. Specifically, it should examine the 'uses' field within the JSON. If 'uses' equals 1, it indicates that the gift code has already been used. Conversely, if 'uses' is 0, the code remains unclaimed and is valid for redemption.

2. **Rate Limiting:** When the tool encounters a 429 HTTP status code, it signifies that rate limiting is in effect. Initially, this results in a brief cooldown period, typically around 10 seconds. However, prolonged usage of the tool can lead to an extended rate limiting duration, often lasting more than a day.

   * **How to Check Rate Limiting:** To determine if rate limiting has been applied, you can perform a simple test. Open your web browser and visit the following link: "https://discordapp.com/api/v9/entitlements/gift-codes/aa?with_application=false&with_subscription_plan=true". If the response states "Unknown gift code," you are not rate limited. However, if it says "You're Rate Limited," you have indeed encountered rate limiting.

**Solution:**

To address these issues effectively, I recommend implementing proxy support within the tool. This feature would allow users to route their requests through various proxies, thus circumventing rate limiting and ensuring the accuracy of gift code status checks.

Proxy support would greatly enhance the tool's functionality and reliability by preventing rate limiting and enabling more precise gift code status assessment.

Thank you for considering this feature request. Please let me know if you have any questions or require further clarification.

**Comment by logicguy1** [comment]
dude... a proxy server only works if it is private and therefore paid, how much fucking money do you have that you want to throw after a nitro generator when you could just fucking buy it.

**Comment by ed6dev** [comment]
> dude... a proxy server only works if it is private and therefore paid, how much fucking money do you have that you want to throw after a nitro generator when you could just fucking buy it.

True, and you have a very very low chance on actually getting a nitro since the links expire + that was a suggestions and there is a lot of free public proxies to use, you can use like [Proxy Scrape](https://proxyscrape.com) Or Any Others. I pay For Mine 2.5$/GB If it's single use 1GB is a lot ( I use it for other stuff ofc ).

**Comment by logicguy1** [comment]
The problem is that most free proxies are already ratelimited as so many people use them
